lib/src/util/flutter_utils.dart

```diff
@@ -61,13 +66,38 @@ class _Flutter {
         _uriContainer = Uri.parse('$uriPrefix/src/widgets/container.dart'),
         _uriFramework = Uri.parse('$uriPrefix/src/widgets/framework.dart');
 
-  bool isExactWidgetTypeContainer(DartType? type) =>
-      type is InterfaceType &&
-      _isExactWidget(type.element, _nameContainer, _uriContainer);
+  bool hasWidgetAsAscendant(ClassElement? element,
+      [Set<ClassElement>? alreadySeen]) {
+    alreadySeen ??= {};
+    if (element == null || alreadySeen.contains(element)) {
```

Comment:
`Set.add` returns `true` if the set was modified, so consider using `!add` in place of `contains`. It's more performant that way.

Reply:
Nice one.  Thanks!